src/HomeNet/Network/ClientList.cs

```diff
@@ -83,6 +78,25 @@ public List<Client> GetNetworkClientList()
     }
 
 
+    /// <summary>
+    /// Gets ID for a new client.
+    /// </summary>
+    /// <returns>New client's network ID.</returns>
+    public ulong GetNewClientId()
+    {
+      log.Trace("()");
+      ulong res = 0;
+
+      lock (lockObject)

```

Comment:
I do that if I see any potential performance benefit. So far this part seems very simple and cheap that a single object was enough for me. I don't think there is a bottleneck here, but if we see any performance problems, this could be measured.